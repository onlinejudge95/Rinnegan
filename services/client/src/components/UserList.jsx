import React from "react";
import PropTypes from "prop-types";

const UserList = (props) => {
  return (
    <div>
      {props.users.map((user) => {
        return (
          <p key={user.id} className="box title is-4 username">
            {user.username}
          </p>
        );
      })}
    </div>
  );
};

UserList.propTypes = {
  users: PropTypes.array.isRequired,
};

export default UserList;
