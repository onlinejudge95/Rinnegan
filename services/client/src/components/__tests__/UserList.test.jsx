import React from "react";
import { render, cleanup } from "@testing-library/react";

import UserList from "../UserList";

afterEach(cleanup);

const userList = [
  { id: 1, username: "test_user_one", email: "test_user_one@mail.com" },
  { id: 2, username: "test_user_two", email: "test_user_two@mail.com" },
];

const mockProps = {
  users: userList,
  removeUser: () => {
    return true;
  },
  isAuthenticated: () => {
    return true;
  },
};

it("renders a username", () => {
  const dom = render(<UserList {...mockProps} />);
  expect(dom.getByText("test_user_one")).toHaveClass("username");
  expect(dom.getByText("test_user_two")).toHaveClass("username");
});

it("renders", () => {
  const dom = render(<UserList {...mockProps} />);
  expect(dom.asFragment()).toMatchSnapshot();
});
