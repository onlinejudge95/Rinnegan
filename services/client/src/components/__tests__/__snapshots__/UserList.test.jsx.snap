// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            Email
          </th>
          <th>
            Username
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            test_user_one@mail.com
          </td>
          <td
            class="username"
          >
            test_user_one
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr>
          <td>
            test_user_two@mail.com
          </td>
          <td
            class="username"
          >
            test_user_two
          </td>
          <td>
            <button
              class="button is-danger is-small"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
